proxy/src/main/java/com/velocitypowered/proxy/bossbar/VelocityBossBar.java

```diff
@@ -0,0 +1,215 @@
+package com.velocitypowered.proxy.bossbar;
+
+import com.velocitypowered.api.bossbar.BossBarColor;
+import com.velocitypowered.api.bossbar.BossBarOverlay;
+import com.velocitypowered.api.proxy.Player;
+import com.velocitypowered.proxy.connection.client.ConnectedPlayer;
+import com.velocitypowered.proxy.protocol.MinecraftPacket;
+import com.velocitypowered.proxy.protocol.packet.BossBar;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import net.kyori.text.Component;
+import net.kyori.text.serializer.gson.GsonComponentSerializer;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class VelocityBossBar implements com.velocitypowered.api.bossbar.BossBar {
+
+  private final List<Player> players;
+  private final UUID uuid;
+  private boolean visible;
+  private Component title;
+  private float progress;
+  private BossBarColor color;
+  private BossBarOverlay overlay;
+  private final VelocityBossBarManager manager;
+
+  public VelocityBossBar(
+      VelocityBossBarManager manager,
+      Component title,
+      BossBarColor color,
+      BossBarOverlay overlay,
+      float progress,
+      UUID uuid) {
+    this.title = title;
+    this.color = color;
+    this.overlay = overlay;
+    this.progress = progress;
+    if (progress > 1 || progress < 0) {
+      throw new IllegalArgumentException("Progress not between 0 and 1");
+    }
+    this.uuid = uuid;
+    visible = true;
+    players = new ArrayList<>();
+    this.manager = manager;
+  }
+
+  private VelocityBossBar() {
```

Comment:
This is a lie. Just junk this constructor and nothing will be tempted to call it.